src/ucp/wireup/select.c

```diff
@@ -383,22 +387,28 @@ static UCS_F_NOINLINE ucs_status_t ucp_wireup_select_transport(
      * best one has the highest score (from the dedicated score_func) and
      * has a reachable tl on the remote peer */
     UCS_BITMAP_FOR_EACH_BIT(tl_bitmap, rsc_index) {
-        resource   = &context->tl_rscs[rsc_index].tl_rsc;
-        iface_attr = ucp_worker_iface_get_attr(worker, rsc_index);
-        md_attr    = &context->tl_mds[context->tl_rscs[rsc_index].md_index].attr;
+        resource_desc  = &context->tl_rscs[rsc_index];
+        resource       = &resource_desc->tl_rsc;
+        iface_attr     = ucp_worker_iface_get_attr(worker, rsc_index);
+        md_attr        = &context->tl_mds[resource_desc->md_index].attr;
+        local_md_flags = criteria->local_md_flags;
 
-        if ((context->tl_rscs[rsc_index].flags & UCP_TL_RSC_FLAG_AUX) &&
+        if ((resource_desc->flags & UCP_TL_RSC_FLAG_AUX) &&
```

Comment:
why needed to change these lines? seems it is not related to p2p connect

Reply:
I've added local helper variable for `resource_desc` to simplify code. It was needed in new added code, and can be used here

Reply:
seems new code uses only rsc_index, worker, and has_cm which do not use resource_desc

Reply:
fixed